test/data_chunk_test.ml

```diff
@@ -0,0 +1,113 @@
+open! Core
+open! Ctypes
+
+(* Tests for Data_chunk module, mimicking DuckDB's data chunk tests *)
+
+let%expect_test "data_chunk_basic" =
+  Duckdb.Database.with_path ":memory:" ~f:(fun db ->
+    Duckdb.Connection.with_connection db ~f:(fun conn ->
+      (* Create a test table *)
+      Duckdb.Query.run_exn' conn {|CREATE TABLE test(a INTEGER, b VARCHAR, c DOUBLE)|};
+      Duckdb.Query.run_exn'
+        conn
+        {|INSERT INTO test VALUES (1, 'hello', 1.5), (2, 'world', 2.5)|};
+      (* Run a query that returns a result *)
+      Duckdb.Query.run_exn conn "SELECT * FROM test" ~f:(fun res ->
+        (* Test data_chunk functionality *)
+        let data_chunk =
+          Duckdb.Result_.fetch res ~f:(fun opt ->
+            match opt with
+            | None -> failwith "Expected data chunk but got None"
+            | Some chunk -> chunk)
+        in
+        (* Test length *)
+        let length = Duckdb.Data_chunk.length data_chunk in
```

Comment:
The logic of checking the length of a data chunk and printing out all columns feels like something that users might want to do. I think functionality similar to `Result_.to_string_hum` should exist for data chunks as well, and we should call that function instead.

Reply:
Call `Data_chunk.to_string_hum` instead?